File: sdks/java/src/main/java/dev/windjammer/sdk/Sprite.java
package dev.windjammer.sdk;

/**
 * 2D sprite component.
 */
public class Sprite {
    private String texture;
    private Vec2 position;
    private Vec2 size;

    /**
     * Creates a new sprite.
     *
     * @param texture Texture path
     * @param position Position
     * @param size Size
     */
    public Sprite(String texture, Vec2 position, Vec2 size) {
        this.texture = texture;
        this.position = position;
        this.size = size;
    }

    public String getTexture() {
        return texture;
    }

    public void setTexture(String texture) {
        this.texture = texture;
    }

    public Vec2 getPosition() {
        return position;
    }

    public void setPosition(Vec2 position) {
        this.position = position;
    }

    public Vec2 getSize() {
        return size;
    }

    public void setSize(Vec2 size) {
        this.size = size;
    }

    @Override
    public String toString() {
        return String.format("Sprite(texture='%s', pos=%s)", texture, position);
    }
}

File: sdks/kotlin/src/main/kotlin/dev/windjammer/sdk/Sprite.kt
package dev.windjammer.sdk

/**
 * 2D orthographic camera.
 */
data class Camera2D(
    var position: Vec2,
    var zoom: Float
) {
    override fun toString() = "Camera2D(pos=$position, zoom=%.2f)".format(zoom)
}

/**
 * 2D sprite component.
 */
data class Sprite(
    var texture: String,
    var position: Vec2,
    var size: Vec2
) {
    override fun toString() = "Sprite(texture='$texture', pos=$position)"
}

// DSL builders
fun camera2D(position: Vec2, zoom: Float = 1.0f) = Camera2D(position, zoom)
fun sprite(texture: String, position: Vec2, size: Vec2) = Sprite(texture, position, size)

